### DISCUSSION TITLE
Transactional Producer, Consumer

### DISCUSSION THREAD

**Discussion by shahabganji** [Farfetch/kafkaflow#553] [Q&A] [answered]
Hi community, I am using Kafkaflow, and faced these two questions: 

1. How could I initiate a transaction on a producer? Like, `producer.inittransaction()` in the Java SKD, I know if I set the `TransactionalId` int the `ProducerConfig` then it is marked as a transactional producer, however when I access the producer via `ProdcuerAccessor.GetProducer()` I could not start a transaction, commit, or rollback! 

2. Is there a possibility to implement, consume-transform-prodcue, in a transactional manner?

PS: I am aware of `TransactionalId` in the `ProducerConfig` and `IsolationLevel = IsolationLevel.ReadCommitted` in the `ConsumerConfig`.

Thanks!

**Comment by joelfoliveira** [comment, marked as answer]
Hi @shahabganji,

Unfortunately, at the moment, KafkaFlow does not support transactions since it doesn't expose the `BeginTransaction`, `CommitTransaction`, or `AbortTransaction` from the Confluent .NET Kafka Driver.
